Fix time support

diegodlh opened the pull request:
I think these commits fix some bugs with time values. Let me know if I should send three separate PRs.

- 3400deb: fixes [validation error](https://github.com/maxlath/wikibase-edit/blob/d42636f20da56c82a55748e3807bb4d7aea419cb/lib/validate.js#L59) because `time` was not considered as a rich value
- 26cf918: fixes getTimeStringBase [error](https://github.com/maxlath/wikibase-edit/blob/d42636f20da56c82a55748e3807bb4d7aea419cb/lib/claim/get_time_object.js#L32) for times with leading plus sign
- 431f206: ISO validation was failing with valid low-precision times such as 2021-00-00 or 2021-01-00.

BTW, regarding 431f206, you [write](https://github.com/maxlath/wikibase-edit/blob/d42636f20da56c82a55748e3807bb4d7aea419cb/lib/datatype_tests.js#L31) "Only trying to parse 5-digit years or below". Shouldn't 5-digit years fail? Shouldn't that be 4-digits only?

maxlath commented:
Ideally, we should have unit tests to avoid future regressions on those different points: would you like to try adding them or should I do it?

On the 5-digit year point: while most cases are limited to 4-digit years, Wikibase supports date far in the past and the future ([example](https://www.wikidata.org/wiki/Q2#P571)), so this lib should probably do what it can to support it

diegodlh commented:
> Ideally, we should have unit tests to avoid future regressions on those different points: would you like to try adding them or should I do it?

Actually, I don't know how to do that. I have to learn, because I'd like to add them to my current and future projects, but I haven't had the time to do that yet :'(

> On the 5-digit year point: while most cases are limited to 4-digit years, Wikibase supports date far in the past and the future (example), so this lib should probably do what it can to support it

Sure! But the [point](https://github.com/maxlath/wikibase-edit/blob/d42636f20da56c82a55748e3807bb4d7aea419cb/lib/datatype_tests.js#L31) I'm referring to is the check you make by converting the time to an ISO string. I understand 5-digit years would fail there (you're already skipping this check for years longer than 5 digits).

maxlath commented:
Learning to use automated tests is definitely a good investment :D I tried to improve the setup documentation to help you give it a try: https://github.com/maxlath/wikibase-edit/blob/master/docs/development_setup.md#development-setup Feel welcome to ask if you get stuck (or would prefer to pass this time ;))

diegodlh commented:
Hi, Max! Thanks for improving the documentation to help me giving it a try! I definitely would like to use this opportunity to learn! More so with a mature project like this, and with a more experienced programmer like you around to help. Although I'm not sure when I will be able to do it. If it can wait a bit more, I'd be thrilled to do it as soon as possible. But feel free to do it yourself if you think it can't wait any longer. Thank you so much!

maxlath commented:
no hurry on my side :)

diegodlh commented:
Hi, @maxlath! I've added some unit tests for the changes introduced by my commits above, as suggested. I've squashed the new commits into the old ones and forced push to keep the history cleaner. When you can, please give it a look and tell me what you think.

> Sure! But the point I'm referring to is the check you make by converting the time to an ISO string. I understand 5-digit years would fail there (you're already skipping this check for years longer than 5 digits).

Regarding this, I'll open a separate issue, to keep things clearer.

Thanks!!

maxlath commented:
nice! published in v4.14.3 :tada:
